pulp_rpm/app/viewsets.py

```diff
@@ -225,3 +208,36 @@ class UpdateRecordViewSet(ContentViewSet):
     serializer_class = UpdateRecordSerializer
     minimal_serializer_class = MinimalUpdateRecordSerializer
     filterset_class = UpdateRecordFilter
+
+
+class OneShotUploadView(views.APIView):
+    """
+    ViewSet for One Shot RPM Upload.
+
+    Args:
+        file@: package to upload
+        repository: repository to update
```

Comment:
It would be great if "repository" were optional.  You could upload a bunch of units without making a bunch of repository versions, and then add them in one operation.